src/logger.cc

```diff
@@ -118,25 +118,18 @@ void Logger::Log(const Napi::CallbackInfo& info) {
 
   if (logger_) {
     std::string message = info[0].As<Napi::String>();
-    switch(level) {
-      case spdlog::level::level_enum::critical:
-        logger_->critical(message);
-        break;
-      case spdlog::level::level_enum::err:
-        logger_->error(message);
-        break;
-      case spdlog::level::level_enum::warn:
-        logger_->warn(message);
-        break;
-      case spdlog::level::level_enum::info:
-        logger_->info(message);
-        break;
-      case spdlog::level::level_enum::debug:
-        logger_->debug(message);
-        break;
-      case spdlog::level::level_enum::trace:
-        logger_->trace(message);
-        break;
+    if constexpr (level == spdlog::level::level_enum::critical) {
+      logger_->critical(message);
+    } else if constexpr (level == spdlog::level::level_enum::err) {
+      logger_->error(message);
+    } else if constexpr (level == spdlog::level::level_enum::warn) {
+      logger_->warn(message);
+    } else if constexpr (level == spdlog::level::level_enum::info) {
+      logger_->info(message);
+    } else if constexpr (level == spdlog::level::level_enum::debug) {
+      logger_->debug(message);
+    } else if constexpr (level == spdlog::level::level_enum::trace) {
+      logger_->trace(message);
```

Comment:
Whats the reason for this change ? We still need to initialize the instance method for each of the enum type so there would be no difference between the two approaches https://godbolt.org/z/5xschjrx9

Reply:
It was to resolve warning [C6326](https://learn.microsoft.com/en-us/cpp/code-quality/c6326?view=msvc-170). Let me know if there's another approach I can take.

Reply:
Okay msvc doesn't optimize for this code https://godbolt.org/z/c84G1E6jb, that's good to know.